### PR TITLE
let user input fileName instead of use DPS

### PR DESCRIPTION
**Pull request by ihere1**

### CODE REVIEW

File: AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp [hunk 1]
```diff
@@ -168,40 +170,35 @@ static int firmwareVerValidated()
...
-	if (badCharCount > 0)
-	{
 		return 1;
 	}
 
 	return 0;
 }
 
-static int fileFullPathValidated(const char * fileFullPath)
+static int fileFullPathValidated(char * fileFullPath)
```
Comment (ArthurMa1978) [inline AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp]: rollback

File: AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp [hunk 2]
```diff
@@ -346,12 +340,44 @@ static unsigned long int findAddressInMapFile(const char * attributeName)
 	return result;
 }
 
+uint8_t* getBinFileWithName(const char *startName, const char *endName, uint32_t &size)
```
Comment (ArthurMa1978) [inline AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp]: good~

File: AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp [hunk 3]
```diff
@@ -346,12 +340,44 @@ static unsigned long int findAddressInMapFile(const char * attributeName)
...
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // App start
 int main()
 {
 	int result = 0;
 
+	fileName = get_user_input("Input the file name of .ino(Click Enter to use \"DPSDemo\" by default): ", fileNameLength);
+	if (strlen(fileName) == 0)
+	{
+		snprintf(fileName, fileNameLength, "DPSDemo");
```
Comment (ArthurMa1978) [inline AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp]: strcpy

File: AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp (hunk 1 above)
Comment (ihere1) [reply AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp]: done.

File: AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp (hunk 3 above)
Comment (ihere1) [reply AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp]: done.

File: AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp [hunk 4]
```diff
@@ -346,12 +340,44 @@ static unsigned long int findAddressInMapFile(const char * attributeName)
...
+	return buf;
+}
+
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // App start
 int main()
 {
 	int result = 0;
 
+	fileName = get_user_input("Input the file name of .ino(Click Enter to use \"DPSDemo\" by default): ", fileNameLength);
```
Comment (ArthurMa1978) [inline AZ3166/tools/dice_device_enrollment/src/dice_device_enrollment/dice_device_enrollment.cpp]: Input the name of your project, default name is \"DPSDemo\" :

**Pull request merged**